service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOController.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.Order;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicListElement;
+import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@Api
+@RestController
+@RequestMapping("/callback/draft-cmo")
+public class DraftCMOController {
+    private final ObjectMapper mapper;
+    private final DraftCMOService draftCMOService;
+
+    @Autowired
+    public DraftCMOController(ObjectMapper mapper, DraftCMOService draftCMOService) {
+        this.mapper = mapper;
+        this.draftCMOService = draftCMOService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        Map<String, Object> caseDataMap = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(caseDataMap, CaseData.class);
+
+        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+
+        DynamicList hearingDatesDynamic = draftCMOService.buildDynamicListFromHearingDetails(hearingDetails);
```

Comment:
We can reach PREPARE FOR HEARING state without hearing details. Should we be able to access this event if there are no hearings to choose from? Showing users an empty list seems unintuitive; perhaps about-to-start validation is needed to check we have at least one hearing date to choose from? Otherwise users starts event and can't progress

Reply:
Yes that's really valid point, Amit can you discuss with Moray what should happen? 

@swalker125 can we issue SDO without Hearing details? should we be able?

Reply:
@akomisarek I discussed it with Moray and he is saying that it should be fine as there will be no hearing date in dropdown and user will not be able to continue.